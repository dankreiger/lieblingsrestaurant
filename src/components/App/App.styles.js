import styled from 'styled-components';

export const AppContainer = styled.div`
  header {
    position: absolute;
    left: 15px;
    top: 1em;
    z-index: 1;
    a {
      padding-right: 10px;
    }
  }
`;
