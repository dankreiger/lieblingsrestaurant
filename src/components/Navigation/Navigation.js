import React from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';
import {
  NavigationContainer,
  FavoritesLink,
  HomeLink,
  RemoveLink
} from './Navigation.styles';
import * as actions from '../../actions';

const Navigation = ({ navigation, toggleNavigation }) => {
  const resetApp = event => {
    event.preventDefault();
    localStorage.removeItem('favoriteRestaurants');
    return (window.location.href = '/'); // find a better way
  };
  return (
    <NavigationContainer
      className={classNames({ toggled: navigation && navigation.toggled })}
    >
      <ul className="sidebar-nav">
        <li className="sidebar-brand">
          <HomeLink to="/" onClick={toggleNavigation}>
            <i className="fas fa-home" />
          </HomeLink>
        </li>
        <li>
          <FavoritesLink to="/info" onClick={toggleNavigation}>
            Favorites
          </FavoritesLink>
        </li>
        <li>
          <RemoveLink to="#" onClick={event => resetApp(event)}>
            Remove All Favorites
          </RemoveLink>
        </li>
      </ul>
    </NavigationContainer>
  );
};

const mapStateToProps = ({ navigation }) => ({ navigation });

export default connect(
  mapStateToProps,
  actions
)(Navigation);
