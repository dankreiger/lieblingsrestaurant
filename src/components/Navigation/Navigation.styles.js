import styled from 'styled-components';
import { Link } from 'react-router-dom';
import {
  NAVIGATION_HEIGHT,
  DEFAULT_TRANSITION_DURATION,
  NAV_SHADOW
} from '../../constants';

export const NavigationContainer = styled.div`
  z-index: 1000;
  position: fixed;
  left: 250px;
  width: 0;
  height: 100%;
  margin-left: -250px;
  overflow-y: auto;
  background: #000000; /* fallback for old browsers */
  background: -webkit-linear-gradient(
    to bottom,
    #434343,
    #000000
  ); /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(
    to bottom,
    #434343,
    #000000
  ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */

  transition: width ${DEFAULT_TRANSITION_DURATION}ms ease,
    box-shadow ${DEFAULT_TRANSITION_DURATION}ms ease
      ${DEFAULT_TRANSITION_DURATION}ms;
  &.toggled {
    width: 250px;
    box-shadow: ${NAV_SHADOW};
  }

  #page-content-wrapper {
    width: 100%;
    position: absolute;
    padding: 15px;
  }

  .sidebar-nav {
    position: absolute;
    top: 0;
    width: 250px;
    height: 100vh;
    margin: 0;
    padding: 0;
    list-style: none;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    font-size: 2.5em;
    padding-top: 75px;
    padding-bottom: 75px;
    li {
      a {
        display: block;
        text-decoration: none;
        color: #ffffff;
        transition: opacity ${DEFAULT_TRANSITION_DURATION}ms,
          transform ${DEFAULT_TRANSITION_DURATION};
        opacity: 1;
        &:hover {
          text-decoration: none;
          opacity: 0.9;
          transform: scale(1.02);
          /* background: rgba(255, 255, 255, 0.2); */
        }
        &:active,
        &:focus {
          text-decoration: none;
        }
      }
    }
  }

  @media (min-width: 768px) {
    width: 0;
    &.toggled {
      width: 250px;
    }
  }
`;

export const HomeLink = styled(Link)`
  display: flex !important;
  font-size: 1.5em;
  align-items: center;
  height: ${NAVIGATION_HEIGHT}px;
`;

export const FavoritesLink = styled(Link)`
  font-size: 1.25em;
`;

export const RemoveLink = styled(Link)`
  color: #ff4136 !important;
  opacity: 1;
  transition: opacity ${DEFAULT_TRANSITION_DURATION};
  &:hover {
    opacity: 0.9;
  }
`;
