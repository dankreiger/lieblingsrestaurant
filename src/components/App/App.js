import React from 'react';
import { Route } from 'react-router-dom';
import Home from '../Home/Home';
import Info from '../Info/Info';
import Navigation from '../Navigation/Navigation';

const App = () => (
  <div>
    <Navigation />
    <main>
      <Route exact path="/" component={Home} />
      <Route exact path="/info" component={Info} />
    </main>
  </div>
);

export default App;
