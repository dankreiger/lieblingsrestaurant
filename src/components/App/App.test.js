import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import Navigation from '../Navigation/Navigation';
import { Route } from 'react-router-dom';

describe('App', () => {
  let appComponent;

  beforeEach(() => {
    appComponent = shallow(<App />);
  });

  describe('rendering', () => {
    test('renders as expected', () => {
      expect(appComponent).toBeTruthy();
      expect(appComponent.length).toBe(1);
      expect(appComponent).toMatchSnapshot();
    });
  });

  describe('structure', () => {
    test('renders 1 <Navigation> component', () => {
      expect(appComponent.find(Navigation)).toBeTruthy();
      expect(appComponent.find(Navigation).length).toBe(1);
      expect(appComponent.find(Route).length).toBe(2);
    });
    test('renders 2 routes', () => {
      expect(appComponent.find(Navigation)).toBeTruthy();
      expect(appComponent.find(Route).length).toBe(2);
    });
  });

  describe('routes', () => {
    test('first route has an exact path leading to the root route', () => {
      expect(
        appComponent
          .find(Route)
          .at(0)
          .props().exact
      ).toBe(true);
      expect(
        appComponent
          .find(Route)
          .at(0)
          .props().path
      ).toBe('/');
    });
    test('second route has an exact path leading to the info route', () => {
      expect(
        appComponent
          .find(Route)
          .at(1)
          .props().exact
      ).toBe(true);
      expect(
        appComponent
          .find(Route)
          .at(1)
          .props().path
      ).toBe('/info');
    });
  });
});
