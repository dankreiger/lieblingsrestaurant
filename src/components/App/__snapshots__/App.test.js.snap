// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(Navigation) />,
        <main>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/info"
          />
        </main>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/info"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/info",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-jTzLTM",
        "isStatic": true,
        "rules": Array [
          "
  /* height: 100vh; */
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-jTzLTM",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(Navigation) />,
          <main>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/info"
            />
          </main>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/info"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/info",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "main",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-jTzLTM",
          "isStatic": true,
          "rules": Array [
            "
  /* height: 100vh; */
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-jTzLTM",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
